core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7470ResolverTransportTest.java

```diff
@@ -0,0 +1,95 @@
+    }
+
+    public void testResolverTransportWagon()
+            throws Exception
+    {
+        File testDir = ResourceExtractor.simpleExtractResources( getClass(), "/mng-7470-resolver-transport" );
+
+        HttpServer server = HttpServer.builder()
+                .port( 0 )
+                .source( new File( testDir, "repo" ) )
+                .build();
+        server.start();
+        try
+        {
+            Verifier verifier = newVerifier( testDir.getAbsolutePath() );
+            HashMap<String, String> properties = new HashMap<>();
+            properties.put( "@port@", Integer.toString( server.port() ) );
+            verifier.filterFile( "settings-template.xml", "settings.xml", "UTF-8", properties );
+
+            verifier = newVerifier( new File( testDir, "wagon" ).getAbsolutePath() );
+            verifier.deleteDirectory( "target" );
+            verifier.deleteArtifacts( "org.apache.maven.its.resolver-transport.wagon" );
+            verifier.addCliOption( "-X" );
+            verifier.addCliOption("-s" );
+            verifier.addCliOption( new File( testDir, "settings.xml" ).getAbsolutePath() );
+            verifier.addCliOption( "-Pmaven-core-it-repo" );
+            verifier.addCliOption( "-Dmaven.resolver.transport=wagon" );
+            // Maven will fail if project dependencies cannot be resolved.
+            // As dependency exists ONLY in HTTP repo, it MUST be reached using selected transport and
+            // successfully resolved from it.
+            verifier.executeGoal( "verify" );
+            verifier.verifyErrorFreeLog();
+            verifier.resetStreams();
+        }
+        finally
+        {
+            server.stop();
+        }
+    }
+
+    public void testResolverTransportNative()
+            throws Exception
+    {
+        File testDir = ResourceExtractor.simpleExtractResources( getClass(), "/mng-7470-resolver-transport" );
+
+        HttpServer server = HttpServer.builder()
+                .port( 0 )
+                .source( new File( testDir, "repo" ) )
+                .build();
+        server.start();
+        try
+        {
+            Verifier verifier = newVerifier( testDir.getAbsolutePath() );
+            HashMap<String, String> properties = new HashMap<>();
+            properties.put( "@port@", Integer.toString( server.port() ) );
+            verifier.filterFile( "settings-template.xml", "settings.xml", "UTF-8", properties );
+
+            verifier = newVerifier( new File( testDir, "native" ).getAbsolutePath() );
+            verifier.deleteDirectory( "target" );
+            verifier.deleteArtifacts( "org.apache.maven.its.resolver-transport.native" );
+            verifier.addCliOption( "-X" );
+            verifier.addCliOption("-s" );
+            verifier.addCliOption( new File( testDir, "settings.xml" ).getAbsolutePath() );
+            verifier.addCliOption( "-Pmaven-core-it-repo" );
+            verifier.addCliOption( "-Dmaven.resolver.transport=native" );
```

Comment:
I can remove it and test still pass ...

Reply:
sure, as it fallbacks to wagon, that's not the point here